src/actions/marketo/marketo_transaction.ts

```diff
@@ -10,21 +10,58 @@ interface Result {
   leads: any[],
   skipped: any[],
   leadErrors: any[],
-  campaignErrors: any[],
+  membershipErrors: any[],
 }
 
 export class MarketoTransaction {
 
   fieldMap: any
   marketo: any
-  campaignId?: string
+  campaignIds: string[] = []
+  addListIds: string[] = []
+  removeListIds: string[] = []
   lookupField?: string
 
   async handleRequest(request: Hub.ActionRequest): Promise<Hub.ActionResponse> {
-
-    this.campaignId = request.formParams.campaignId
-    if (!this.campaignId) {
-      throw "Missing Campaign ID."
+    this.campaignIds = []
+    this.addListIds = []
+    this.removeListIds = []
+    const subactionIds =
+      (request.formParams.subactionIds !== undefined ? request.formParams.subactionIds : "")
+      .split(/\s*,\s*/)
+      .filter(Boolean)
+    switch (request.formParams.subaction) {
+      case undefined:
+        // The older version of the action assumed an "addToCampaign" subaction
+        this.campaignIds = [request.formParams.campaignId !== undefined ? request.formParams.campaignId : ""]
+          .filter(Boolean) // Note singular parameter name
+        break
```

Comment:
@phillipperalez this logic is critical to make sure that existing schedules that were created with the existing version of the Marketo action do not break.  It defaults to the update record and add to campaign action, which is the only one that the current version supports.